app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java

```diff
@@ -0,0 +1,240 @@
+package nyc.c4q.jonathancolon.inContaq.utlities.graphs;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.animation.BounceInterpolator;
+
+import com.db.chart.Tools;
+import com.db.chart.animation.Animation;
+import com.db.chart.model.LineSet;
+import com.db.chart.view.LineChartView;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.ExecutionException;
+
+import nyc.c4q.jonathancolon.inContaq.R;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.Sms;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.monthly.MonthlyReceivedWorkerTask;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.monthly.MonthlySentWorkerTask;
+import nyc.c4q.jonathancolon.inContaq.utlities.sms.monthly.MonthlyTaskParams;
+
+import static android.graphics.Color.parseColor;
+import static com.db.chart.renderer.AxisRenderer.LabelPosition.NONE;
+
+public class WeeklyGraph {
+
+    private static final String BLUE_SAPPHIRE = "#0E587A";
+    private static final String BLUE_MASTRICHT = "#02283A";
+    private static final String RED_ROSE_MADDER = "#E71D36";
+    private static final String WHITE_BABY_POWDER = "#FDFFFC";
+    private static final String YELLOW_CRAYOLA = "#FF9F1C";
+    private static final String MONTHLY_SENT = "Monthly Sent: ";
+    private static final String MONTHLY_RECEIVED = "Monthly Received: ";
+    private static final int SUN = 1;
+    private static final int MON = 2;
+    private static final int TUE = 3;
+    private static final int WED = 4;
+    private static final int THUR = 5;
+    private static final int FRI = 6;
+    private static final int SAT = 7;
+    private static final int DEFAULT_VALUE = 0;
+    private final String TAG = "sms";
+    private Context context;
+    private int highestValue;
+    private LineChartView lineGraph;
+    private float[] receivedValues;
+    private float[] sentValues;
+    private ArrayList<Sms> lstSms;
+
+    public WeeklyGraph(Context context, LineChartView lineGraph, ArrayList<Sms> lstSms) {
+
+        this.context = context;
+        this.lineGraph = lineGraph;
+        this.lstSms = lstSms;
+    }
+
+    public void showWeeklyGraph() {
+
+        getLineGraphValues(lstSms);
+        getHigehstValueForYaxis();
+        loadGraph();
+    }
+
+    private void getLineGraphValues(ArrayList<Sms> lstSms) {
+
+        receivedValues = setValues(getDailyReceived(lstSms));
+        sentValues = setValues(getMonthlySent(lstSms));
+    }
+
+    private float[] setValues(TreeMap<Integer, Integer> numberOfTexts) {
```

Comment:
yea you could try doing something like getDailyReceived(listSms).values().toArray(); Also setValues looks like it's really a getter method.